todo_app/templates/index.html

```diff
@@ -7,12 +7,65 @@ <h1 class="display-4">To-Do App</h1>
     <p class="lead">Just another to-do app.</p>
   </div>
 
+  <div class="row mb-4">
+    <div class="col-md-12">
+      <h2>Add Item</h2>
+      <form action="{{ url_for('index') }}" method="post">
+          <div class="form-row">
+            <div class="form-group col-md-8">
+              <input type="text" class="form-control" name="title" placeholder="Enter a task here" required>
+            </div>
+            <div class="form-group col-md-3">
+              <button type="submit" class="btn btn-success btn-block">Submit</button>
+            </div>
+          </div>
+      </form>
+    
+      {% if error %}
+        <p class="text-danger">{{ error }}</p>
+      {% endif %}
+    </div>
+  </div>
+
   <div class="row">
     <div class="col-md-12">
       <h2>Items</h2>
-      <ul class="list-group mb-4">
-        <!-- List the todo items here -->
-      </ul>
+      <table class="table table-bordered mb-4">
+        <thead class="thead-dark">
+          <tr>
+            <th>#</th>
+            <th>Title</th>
+            <th>Status</th>
+            <th>Actions</th>
+          </tr>
+        </thead>
+        <tbody>
+          {% for index in range(0, todos|length) %}
+              {% set todo = todos[index] %}
```

Comment:
Nice solution to this! Jinja provides us with an inbuilt way to grab the index from the loop though, so we could equally do the below, which might be a little clearer
```python
{% for todo in todos %}
              <tr>
                <td>{{loop.index}}</td>
                <td>{{todo.title}}</td>
... 
```

Reply:
Cheers Jack, done.